### ISSUE TITLE
llvm-project file causes tag parser crash (Warning: Database safety check failed. Recreating database)

### ISSUE THREAD

**Issue by Bestie123** [microsoft/vscode-cpptools#9679]
### Environment

- OS and version: windows 10
- VS Code: 1.69.2
- C/C++ extension: 1.11.4 

### Bug Summary and Steps to Reproduce

Bug Summary:
- cpptools: Connection to server got closed. Server will not be restarted.
- c/c++ : Warning: Database safety check failed. Recreating database.
directory used as a project:
https://github.com/llvm/llvm-project

1) you can download it only because an error occurs when analyzing this file
https://github.com/llvm/llvm-project/blob/main/libcxx/test/support/test.support/make_string_header.pass.cpp

2) open folder in vs code
3) automatic analize 
4) error

### Expected behavior

when installing c/c++ 1.10.8, the project analyzes normally and no errors appear

### Code sample and Logs

```shell
c_cpp_properties.json


{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "windowsSdkVersion": "10.0.19041.0",
            "compilerPath": "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.32.31326/bin/Hostx64/x64/cl.exe",
            "cStandard": "c17",
            "cppStandard": "c++14",
            "intelliSenseMode": "windows-msvc-x64"
        }
    ],
    "version": 4
}
```

### Screenshots

- C/C++ extension: 1.11.4  output to the console when analyzing a file when an error occurs, and start re-analyzing the project.

G:\2\libcxx\test\support\test.support\make_string_header.pass.cpp
![image](https://user-images.githubusercontent.com/15684883/182661703-3702c955-d66a-4213-9e50-8fd40a1d07fd.png)

cpptools console error
![image](https://user-images.githubusercontent.com/15684883/182666445-5dce99fa-e969-4ba6-8e2d-de64a2e828d2.png)

when installing c/c++ 1.10.8, the project analyzes normally and no errors appear, a message about the completion of the analysis is displayed in the console

### Additional context

[make_string_header.pass.zip](https://github.com/microsoft/vscode-cpptools/files/9253612/make_string_header.pass.zip)

_No response_

**Comment by michelleangela** [comment]
The contents of the file https://github.com/llvm/llvm-project/blob/main/libcxx/test/support/test.support/make_string_header.pass.cpp cause the language server to crash.

**Comment by Colengms** [comment]
@Bestie123 Is it possible the crash was caused by? : `clang\test\parser\parser_overflow.c` https://github.com/microsoft/vscode-cpptools/issues/9653

Without that file, I'm able to parse all of the latest code currently in the llvm-project repo.  (Though, perhaps there was recent change that removed your repro?).

**Comment by Bestie123** [comment]
@Colengms 
https://github.com/llvm/llvm-project/releases
LLVM 14.06
LLVM 15.0.0-rc1
also tried the current latest copy of the repository

clang\test\parser\parser_overflow.c  analyzed without errors

tested on versions of c/c++:  1.10.8,  pre release version (1.12.0, 1.11.4, 1.11.3)

**but I did not run the analysis of the entire project, but only the clang\test\parser directory**

![image](https://user-images.githubusercontent.com/15684883/182996218-f470e577-61a4-4b60-96ed-9046b58ee51b.png)

**Comment by Colengms** [comment]
Hi @Bestie123 .  Could you try using the 1.12.0 pre-release version of cpptools?  I suspect that perhaps the issue you are seeing may have been addressed in 1.12.0, though an issue repro'ing with `parser_overflow.c` may have been introduced in that version.

**Comment by Bestie123** [comment]
@Colengms  I tried to run the analysis libcxx\test\support\test.support\make_string_header.pass.cpp on cpptools version 1.12.0, but the analysis also gives an error
![image](https://user-images.githubusercontent.com/15684883/183242328-ad87b06f-d8be-4c76-a22b-8c96f3ee087c.png)

**Comment by Bestie123** [comment]
@sean-mcmanus 
the error in version 1.12.0 no longer appears during analysis
libcxx\test\support\test.support\make_string_header.pass.cpp
![image](https://user-images.githubusercontent.com/15684883/183740637-f6c73111-9ce0-404e-aa04-bd0f3f41bc8d.png)

**Comment by sean-mcmanus** [comment]
FYI, 1.11.5 and 1.12.1 have a workaround for this (avoid tag parsing that file), but 1.12.2 should have a real fix.

You may be seeing it fixed in 1.12.0 because it may have saved the parsed state in the database, but it would still crash if the database were reset or if the file was edited/saved.
